### PR TITLE
Fix `logger.exception` call

### PR DESCRIPTION
**Pull request by harupy**
[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/9785?quickstart=1)

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Fix for https://github.com/mlflow-automation/mlflow/actions/runs/6370871713/job/17294713645#step:13:13513:

```
self = <LogRecord: test_spark_model_export, 40, /home/runner/work/mlflow/mlflow/tests/spark/test_spark_model_export.py, 81, "[JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.">

    def getMessage(self):
        """
        Return the message for this LogRecord.
    
        Return the message for this LogRecord after merging any user-supplied
        arguments with the message.
        """
        msg = str(self.msg)
        if self.args:
>           msg = msg % self.args
E           TypeError: not all arguments converted during string formatting

msg        = '[JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.'
self       = <LogRecord: test_spark_model_export, 40, /home/runner/work/mlflow/mlflow/tests/spark/test_spark_model_export.py, 81, "[JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.">
```

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/spark/test_spark_model_export.py [hunk 1]
```diff
@@ -75,12 +75,10 @@ def _get_spark_session_with_retry(max_tries=3):
...
         try:
             return get_spark_session(conf)
-        except Exception as e:
+        except Exception:
             if num_tries >= max_tries - 1:
                 raise
-            _logger.exception(
-                e, f"Attempt {num_tries} to create a SparkSession failed, retrying..."
-            )
+            _logger.exception(f"Attempt {num_tries} to create a SparkSession failed, retrying...")
```
Comment (harupy) [inline tests/spark/test_spark_model_export.py:81]: The first argument should be a string, not an exception.

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Comment by harupy** [comment]
@WeichenXu123 This PR just fixes the logger.exception call, and doesn't fix the original error.

**Comment by harupy** [comment]
`TypeError: not all arguments converted during string formatting` is fixed. Merging.

**Pull request merged**